framework/include/base/NonlinearEigenSystem.h

```diff
@@ -62,6 +62,15 @@ class NonlinearEigenSystem
 
   virtual NumericVector<Number> & RHS() override;
 
+  // return the Nth converged eigenvlue
+  virtual std::pair<Real, Real>  getNthConvergedEigenvalue(dof_id_type n) { return _transient_sys.get_eigenpair(n); }
```

Comment:
This is calling down into libMesh and libMesh is probably doing bounds checking. If not, we should assert that n is in range. Can you verify? Method should be `const`